### PR TITLE
feat(BDropdown): add variant, classes and correct attrs to text sub components

### LINKED ISSUE #2244: Several of the sub-components of BDropdown are not fully implemented
### Missing feature is from

bootstrap-vue

### Docs

https://bootstrap-vue.org/docs/components/dropdown#dropdown-supported-sub-components

### Describe the missing feature

Three of the helper components for `BDropdown` don't implement any of the props described in the BSV documentation. They are
- `BDropdownForm`
- `BDropdownHeader`
- `BDropdownText`

### Validations

- [X] Have tested with the latest version. This is still alpha version and sometime things change rapidly.
- [X] Follow our [Code of Conduct](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/bootstrap-vue-next/bootstrap-vue-next/blob/main/CONTRIBUTING.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided link is to the documentation of the missing feature.

### PR DESCRIPTION
**Pull request by xvaara**
# Describe the PR

closes #2244

## Small replication

A small replication or video walkthrough can help demonstrate the changes made. This is optional, but can help observe the intended changes. A mentioned issue that contains a replication also works.

## PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix :bug: - `fix(...)`
- [x] Feature - `feat(...)`
- [ ] ARIA accessibility - `fix(...)`
- [ ] Documentation update - `docs(...)`
- [ ] Other (please describe)

**The PR fulfills these requirements:**

- [x] Pull request title and all commits follow the [**Conventional Commits**](https://www.conventionalcommits.org/) convention or has an [**override**](https://github.com/googleapis/release-please#how-can-i-fix-release-notes) in this pull request body **This is very important, as the `CHANGELOG` is generated from these messages, and determines the next version type. Pull requests that do not follow conventional commits or do not have an override will be denied**

### CODE REVIEW

**Comment by dwgray** [comment]
In BSV, the `innerClass`, when implemented, was more specific - e.g. `formClass`, `textClass`, etc. I could argue either way for doing that vs. using the more generic term in all cases. I'm happy whichever way you want to go. But I do strongly prefer that we maintain internal consistency. If you want to make everything `innerClass`, that would require changing basically all the dropdown sub-components you didn't touch here from their specific names to `innerClass`, or to go the other direction use the same prop names as BSV used. @VividLemon , do you have any thoughts on this?

File: packages/bootstrap-vue-next/src/components/BDropdown/BDropdownDivider.vue [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
 </template>
 
 <script setup lang="ts">
+import {useAttrs} from 'vue'
 import {useDefaults} from '../../composables/useDefaults'
 import type {BDropdownDividerProps} from '../../types/ComponentProps'
 
+defineOptions({
+  inheritAttrs: false,
+  innerClass: undefined,
```
[the remark this answers is not in the document]
Comment (VividLemon) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdownDivider.vue:21]: Looks odd I agree

[the remark this answers is not in the document]
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/BDropdownDivider.vue:21]: ups, that was a mistake

File: packages/bootstrap-vue-next/src/components/BDropdown/dropdown-text.spec.ts [hunk 2]
```diff
@@ -33,19 +33,4 @@ describe('dropdown-text', () => {
     })
     expect(wrapper.text()).toBe('foobar')
   })
-
-  it('renders prop text', () => {
```
Comment (VividLemon) [inline packages/bootstrap-vue-next/src/components/BDropdown/dropdown-text.spec.ts:37]: Why is this removed

**Review by VividLemon — CHANGES_REQUESTED** [review]

File: packages/bootstrap-vue-next/src/components/BDropdown/dropdown-text.spec.ts (hunk 2 above)
Comment (xvaara) [reply packages/bootstrap-vue-next/src/components/BDropdown/dropdown-text.spec.ts:37]: At some point I removed the test prop.. I'll return the test

**Comment by dwgray** [comment]
@xvaara I believe that we should implement `novalidate` and `validated` on `BDropdownForm` - these are implemented on `BForm` and `BNavForm`, so it would be weird not to have them on `BDropdownForm`.  I've got a commit with these changes https://github.com/bootstrap-vue-next/bootstrap-vue-next/commit/7b51923355ba4e0216e95b0f31d54dd1235e1a09 - I don't see a way to apply my commit to your PR, but if you can do it or point me at how to do it that would be great.

I'd like to mark `disabled` on `BDropdownForm` as deprecated - in BSV, this appears to disable all of the controls on the form. This is _not implemented_ on `BForm` or `BNavForm` in BSV. This is certainly something that could be added after a v1 in a non-breaking way if someone feels it's necessary.

**Comment by xvaara** [comment]
@dwgray I added the stuff from that commit to this.

**Review by dwgray — APPROVED** [review]
LGTM with the exception of the duplicate test.

**Pull request merged**